src/panels/logbook/ha-logbook.ts

```diff
@@ -194,7 +193,13 @@ export class HaLogbook extends LitElement {
 
   private _unsubscribe(): void {
     if (this._subscribed) {
-      this._subscribed.then((unsub) => (unsub ? unsub() : undefined));
+      this._subscribed
+        .then((unsub) => (unsub ? unsub() : undefined))
+        .catch(() => {
```

Comment:
Since this is an API implementation detail, we should move it inside `subscribeLogbook`. It would just not return an `unsub` if it was canceled ?

Reply:
The unsub being undefined is because of the promise rejection here https://github.com/home-assistant/frontend/blob/91cd584b4b5dd57392f42aac7ee2fe9b5b7255c7/src/data/logbook.ts#L187

The catch is for the throw when we get when we try to unsubscribe the and the subscription is already gone because the backend/core canceled it because everything we subscribed to was filtered.

Reply:
Ooh I see it now. Damn this is all confusing.

So `unsubscribed` already catches all exceptions:

https://github.com/home-assistant/frontend/blob/dev/src/panels/logbook/ha-logbook.ts#L266-L269

So this `catch` cannot happen it seems?

Reply:
The catch can happen in this case:

1. unsub is there (the guard in `subscribeLogbook` only knows about missing data not state_classes / UOM / etc reasons why the backend will reject the subscription to an empty set of entities) 
2. unsub gets called but since the backend has already canceled the subscription which needs to be caught otherwise it raises.

Reply:
Aah, ok. So it's catching exceptions caused by calling `unsub()`. `unsub()` will send a message to the server to unsubscribe ([source](https://github.com/home-assistant/home-assistant-js-websocket/blob/master/lib/connection.ts#L328-L335)). It feels like a bad design part that the frontend doesn't know they are no longer subscribed. This is however an issue in the WebSocket API and not the concern of this PR.

Can we move the `.catch()` on the `unsub` to be clear what is going on here? `unsub().catch(…)`

Reply:
> Aah, ok. So it's catching exceptions caused by calling `unsub()`. `unsub()` will send a message to the server to unsubscribe ([source](https://github.com/home-assistant/home-assistant-js-websocket/blob/master/lib/connection.ts?rgh-link-date=2022-06-06T22%3A49%3A40Z#L328-L335)). It feels like a bad design part that the frontend doesn't know they are no longer subscribed. This is however an issue in the WebSocket API and not the concern of this PR.

I agree this is clunky.  I thought about fixing it by having the backend keep the subscription until the frontend cancels it even if it is currently not doing anyways.  Another possible solution would be to send something in the metadata to tell the frontend goodbye you asked for something I can't actually deliver.

> 
> Can we move the `.catch()` on the `unsub` to be clear what is going on here? `unsub().catch(…)`

Testing that now

Reply:
<img width="540" alt="Screen Shot 2022-06-06 at 13 06 46" src="https://user-images.githubusercontent.com/663432/172263597-0a008dbb-e155-4b99-bf7b-099825d21b34.png">

Reply:
Oh. unsub isn't a promise its the actual unsubscribe function
`export declare type UnsubscribeFunc = () => void;`

Reply:
The type `UnsubscribeFunc` was wrong. The actual type of the unsubscribe function is `() => Promise<void>`

Reply:
Well that will do it 👍 

Retesting now

Reply:
Passing with my logbook card torture test dashboard 👍

Reply:
All good on the production instances that have enough memory to run gulp as well